### PR TITLE
fix(compiler-cli): emit correct css string escape sequences

### LINKED ISSUE #22774: Literals in css with escapes are not emitted correctly
## I'm submitting a...

<pre><code>
[x] Regression (a behavior that used to work and stopped working in a new release)
</code></pre>

REF: [b/74538140](http://b/74538140)

## Current behavior

A style that contains an a value with a string escape such as `'\0025BC'` is emitted as `"\\x0025BC"` instead of the expected "\0025BC".

This is true when using 2.6 or 2.7 of TypeScript but does not reproduce with 2.5.

This is related to https://github.com/Microsoft/TypeScript/issues/20192. TypeScript, beginning with 2.6, calls `escapeString` for the value of synthetic literals.

https://github.com/Microsoft/TypeScript/blob/1736741e546bb6d436acd4c0632e3b23c7033b29/src/compiler/utilities.ts#L352

This produces the incorrect escaping identically had this been an interpolation.

## Expected behavior

The escaped string in css should be emitted correctly.

## Minimal reproduction of the problem with instructions

1. `ng new repro`
2. modify app.component.css to be

```css
span {
  content: '\0025BC';
}
```

`./node_modules/.bin/ngc -p src/tsconfig.app.json`

Inspect `out-tsc/app/src/app/app.component.css.shim.ngstyle.js`

**EXPECTED**

```
["span[_ngcontent-%COMP%] {\n  content: '\\0025BC';\n}"]
```

**RECEIVED**

```
["span[_ngcontent-%COMP%] {\n  content: '\\x00025BC';\n}"]
```

### PR DESCRIPTION
**Pull request by chuckjaz**
Works around an issue with TypeScript 2.6 and 2.7 that causes
the tranformer emit to emit incorrect escapes for css string
literals.

Fixes: #22774

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
```

## What is the current behavior?

Issue Number: #22774

## What is the new behavior?

Bypasses TypeScript incorrect string escaping to produce the correct `.js` file.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by vicb — APPROVED** [review]
Congrats on the terrible hack :)

File: packages/compiler-cli/src/transformers/node_emitter.ts [hunk 1]
```diff
@@ -150,13 +150,28 @@ function firstAfter<T>(a: T[], predicate: (value: T) => boolean) {
 // NodeEmitterVisitor
 type RecordedNode<T extends ts.Node = ts.Node> = (T & { __recorded: any; }) | null;
 
+function escapeLiteral(value: string): string {
+  return value.replace(/(\"|\\)/g, '\\$1').replace(/(\n)|(\r)/g, function(v, n, r) {
```
Comment (vicb) [inline packages/compiler-cli/src/transformers/node_emitter.ts]: would `.replace(\n).replace(\r)` be more readable ?

Comment (vicb) [reply packages/compiler-cli/src/transformers/node_emitter.ts]: or `.replace(/(\n|\r)/)` ?

**Review by vicb — APPROVED** [review]

**Review by mhevery — APPROVED** [review]

File: packages/compiler-cli/src/transformers/node_emitter.ts (hunk 1 above)
Comment (chuckjaz) [reply packages/compiler-cli/src/transformers/node_emitter.ts]: Sure but slower.

**Pull request closed without merging**